### PR TITLE
Update SignalR Cmdlets

### PR DESCRIPTION
**Pull request by Tashi711**
## Description

1. Add "Update", "CheckName", "Restart", "GetUsage" Cmdlets for SignalR.
2. Fix bug of no change when executing "New" Cmdlet for updating SignalR resource.
3. Update version of SDK and the corresponding parameters (e.g. Feature)

## Checklist

- [x] I have read the [_Submitting Changes_](../CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [X] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by cormacpayne** [comment]
Approved cmdlet design review: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/356

**Comment by cormacpayne** [comment]
@isra-fel let me know if there's anything else you find in the code review 😁

File: src/SignalR/SignalR/Cmdlets/UpdateAzureRmSignalR.cs [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+            Position = 0,
+            ParameterSetName = ResourceGroupParameterSet,
+            HelpMessage = "The SignalR service name.")]
+        [ValidateNotNullOrEmpty()]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipeline = true,
```
[the remark this answers is not in the document]
Comment (Tashi711) [reply src/SignalR/SignalR/Cmdlets/UpdateAzureRmSignalR.cs]: fixed

File: src/SignalR/SignalR/Cmdlets/RestartAzureRmSignalR.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            Position = 0,
+            ParameterSetName = ResourceGroupParameterSet,
+            HelpMessage = "The SignalR service name.")]
+        [ValidateNotNullOrEmpty()]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipeline = true,
```
[the remark this answers is not in the document]
Comment (Tashi711) [reply src/SignalR/SignalR/Cmdlets/RestartAzureRmSignalR.cs]: fixed

File: src/SignalR/SignalR/Az.SignalR.psd1 [hunk 3]
```diff
@@ -110,7 +111,7 @@ PrivateData = @{
         # IconUri = ''
 
         # ReleaseNotes of this module
-        ReleaseNotes = '* Fixed miscellaneous typos across module'
+        ReleaseNotes = '* Add Update, Restart, CheckNameAvailability, GetUsage Cmdlets'
```
[the remark this answers is not in the document]
Comment (Tashi711) [reply src/SignalR/SignalR/Az.SignalR.psd1]: fixed

File: src/SignalR/SignalR/ChangeLog.md [hunk 4]
```diff
@@ -19,6 +19,9 @@
 -->
 ## Upcoming Release
 
+## Version 1.0.4
```
[the remark this answers is not in the document]
Comment (Tashi711) [reply src/SignalR/SignalR/ChangeLog.md]: fixed

File: src/SignalR/SignalR/Cmdlets/RestartAzureRmSignalR.cs [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+using Microsoft.Azure.Management.SignalR;
+
+namespace Microsoft.Azure.Commands.SignalR.Cmdlets
+{
+    [Cmdlet("Restart", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SignalR", SupportsShouldProcess = true, DefaultParameterSetName = ResourceGroupParameterSet)]
+    [OutputType(typeof(bool))]
+    public class RestartAzureRmSignalR : SignalRCmdletBase, IWithInputObject, IWithResourceId
+    {
+        [Parameter(
+            Mandatory = false,
```
[the remark this answers is not in the document]
Comment (Tashi711) [reply src/SignalR/SignalR/Cmdlets/RestartAzureRmSignalR.cs]: @cormacpayne
Using a default `ResourceGroupName` is for consistance with existed cmdlets which use `ResourceGroupName` as a parameter. Most cmdlets of SignalR require users to provide `ResourceGroupName` parameter (`New-AzSignalR`, `Get-AzSignalR`, `Get-AzSignalRKey`, `New-AzSignalRKey`, `Remove-AzSignalR` and the new-added `Update-AzSignalR`, `Restart-AzSignalR`). In most situation, users will run these cmdlets within a same resource group. So setting a default resource group is much convenient for users to run these cmdlets, not necessary to provide a same `ResourceGroupName` each time.

File: src/SignalR/SignalR/Cmdlets/NewAzureRmSignalR.cs [hunk 6]
```diff
@@ -53,88 +55,86 @@ public sealed class NewAzureRmSignalR : SignalRCmdletBase
...
 
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The service mode for the SignalR service.")]
+        [PSArgumentCompleter("Default", "Serverless", "Classic")]
+        public string ServiceMode { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The allowed origins for the SignalR service. To allow all, use \"*\" and remove all other origins from the list. Slashes are not allowed as part of domain or after TLD")]
```
Comment (isra-fel) [inline src/SignalR/SignalR/Cmdlets/NewAzureRmSignalR.cs]: Can we use avoid using abbr? Won't want our customers to guess what TLD means...

File: src/SignalR/SignalR/Cmdlets/UpdateAzureRmSignalR.cs [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The service mode for the SignalR service.")]
+        [PSArgumentCompleter("Default", "Serverless", "Classic")]
+        public string ServiceMode { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The allowed origins for the SignalR service. To allow all, use \"*\" and remove all other origins from the list. Slashes are not allowed as part of domain or after TLD")]
```
Comment (isra-fel) [inline src/SignalR/SignalR/Cmdlets/UpdateAzureRmSignalR.cs]: same here

File: src/SignalR/SignalR/Cmdlets/NewAzureRmSignalR.cs (hunk 6 above)
Comment (Tashi711) [reply src/SignalR/SignalR/Cmdlets/NewAzureRmSignalR.cs]: @isra-fel

fixed

File: src/SignalR/SignalR/Cmdlets/UpdateAzureRmSignalR.cs (hunk 7 above)
Comment (Tashi711) [reply src/SignalR/SignalR/Cmdlets/UpdateAzureRmSignalR.cs]: @isra-fel

fixed

File: src/SignalR/SignalR/Cmdlets/UpdateAzureRmSignalR.cs [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+{
+    [Cmdlet("Update", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SignalR", SupportsShouldProcess = true, DefaultParameterSetName = ResourceGroupParameterSet)]
+    [OutputType(typeof(PSSignalRResource))]
+    public class UpdateAzureRmSignalR : SignalRCmdletBase, IWithInputObject, IWithResourceId
+    {
+        private const string DefaultSku = "Standard_S1";
+        private const int DefaultUnitCount = 1;
+
+        [Parameter(
+            Mandatory = false,
```
[the remark this answers is not in the document]
Comment (Tashi711) [reply src/SignalR/SignalR/Cmdlets/UpdateAzureRmSignalR.cs]: @cormacpayne
same comment here as above.

**Pull request merged**